### ISSUE TITLE
RPC command line on windows

### ISSUE THREAD

**Issue by AtmanActive** [mumble-voip/mumble#5909]
### The issue

Hi,

I'm trying to use RPC command line to automate running Mumble instance on windows but it doesn't work.
I have [MumblePortable](https://portableapps.com/apps/internet/mumble-portable) running and I issue:

`mumble.exe rpc mute`

... and nothing happens.

When I issue

`mumble.exe -h`

... then I do get the window with all command line options.

I read [here](https://github.com/StarGate01/AmongUs-Mumble/issues/53) someone suggested to check the path in the config file. I do have mumble.ini but I don't know what syntax would that be.

Any advice would be highly appreciated.

Thanks.

### Mumble version

1.4.274

### Mumble component

Client

### OS

Windows

**Comment by Krzmbrzl** [comment]
The first thing to try would be to check if a regular Mumble install shows the same issue. We don't provide portable installations (yet) and thus I don't know what they did to produce a portable Mumble version. Maybe this involves some sandboxing that also prevents RPC from working :thinking:

**Comment by AtmanActive** [comment]
I just tested with the latest official version and it is working as expected.

Even though I didn't even install the MSI file but just unpacked it and ran mumble.exe from there.

So what's different?
How does Mumble determine where to direct the RPC command?

As far as I know, MumblePortable doesn't touch mumble.exe itself in any way. It just prepares the working dir and config files and runs mumble.exe.

Will investigate some more...

**Comment by AtmanActive** [comment]
This is how mumble,exe is ran by MumblePortable.exe:

`mumble.exe /UAC:79283A /NCRC`

Could this be the culprit?

I have no idea what those arguments are.

**Comment by Krzmbrzl** [comment]
> I have no idea what those arguments are.

Me neither. Those are not handled by us.

In terms of how RPC is handled, I believe this is handled by Qt. Not quite sure though...

**Comment by AtmanActive** [comment]
OK, got it figured out.

First I tried running the official mumble.exe with `/UAC:79283A /NCRC` arguments - no change, it is still working as expected.

Then I found `MumblePortable.ini` in the subfolder `App\AppInfo\Launcher` of the MumblePortable folder structure and there, on line 6 there was `RunAsAdmin=try`. When I removed that line, mumble RPC started working as expected.

On the other hand, if I leave that line `RunAsAdmin=try` in and I run my windows command prompt as admin, then, again, RPC is working correctly.

So the issue was admin elevation.
If mumble,exe is ran as admin, then, one needs a windows command prompt ran as admin as well in order for RPC to work.
If mumbe.exe is not elevated, then, a normal command prompt can reach RPC.
In other words non-elevated mumble.exe-RPC can't reach an elevated mumble.exe. Which is to be expected.

Now the question remains: does mumble.exe really need to be run as admin?

Or was this just an oversight by PortableApps.com packers?

I don't mind either, but would like to know what is official Mumble's position on this.

Thanks.

**Comment by davidebeatrici** [comment]
Mumble should never be run as administrator, but there is a potential reason for doing so: getting global shortcuts to work in privileged applications.

We make use of `uiAccess` in official releases:

https://github.com/mumble-voip/mumble/blob/98fb1cd8b42ce5020a9696c571ca470fe7cd569b/src/mumble_exe/CMakeLists.txt#L35-L37

From https://learn.microsoft.com/en-us/windows/security/threat-protection/security-policy-settings/user-account-control-allow-uiaccess-applications-to-prompt-for-elevation-without-using-the-secure-desktop:

> UIA programs are designed to interact with Windows and application programs on behalf of a user. This setting allows UIA programs to bypass the secure desktop to increase usability in certain cases, but it allows elevation requests to appear on the regular interactive desktop instead of on the secure desktop. This requests-appearance increases the risk that a malicious program could intercept data that is being transferred between the UI and the application. Because UIA programs must be able to respond to prompts regarding security issues, such as the UAC elevation prompt, UIA programs must be highly trusted. To be considered trusted, a UIA program must be digitally signed. By default, UIA programs can be run only from the following protected paths:
>
> - ..\Program Files\ (and subfolders)
> - ..\Program Files (x86)\ (and subfolders, in 64-bit versions of Windows only)
> - ..\Windows\System32\
>
> The requirement to be in a protected path can be disabled by the [User Account Control: Only elevate UIAccess applications that are installed in secure locations](https://learn.microsoft.com/en-us/windows/security/threat-protection/security-policy-settings/user-account-control-only-elevate-uiaccess-applications-that-are-installed-in-secure-locations) setting. Although this setting applies to any UIA program, it's used primarily in certain Windows Remote Assistance scenarios.

**Comment by Krzmbrzl** [comment]
As Davide said. The gist of it is: At this point you either install Mumble in the programs directory on your system or global shortcuts will only work while not focusing on an application that runs with raised privileges.
